pinot-common/src/main/java/org/apache/pinot/common/request/context/ExpressionContext.java

```diff
@@ -31,37 +34,53 @@
  */
 public class ExpressionContext {
   public enum Type {
-    LITERAL, IDENTIFIER, FUNCTION
+    LITERAL, IDENTIFIER, FUNCTION,
   }
 
   private final Type _type;
   private final String _value;
   private final FunctionContext _function;
+  // Only set when the _type is LITERAL
+  @Nullable
+  private final LiteralContext _literal;
 
-  public static ExpressionContext forLiteral(String literal) {
-    return new ExpressionContext(Type.LITERAL, literal, null);
+  public static ExpressionContext forLiteralContext(Literal literal){
+    return new ExpressionContext(Type.LITERAL, null, null, new LiteralContext(literal));
+  }
+  public static ExpressionContext forLiteralContext(FieldSpec.DataType type, Object val){
+    return new ExpressionContext(Type.LITERAL, null, null, new LiteralContext(type, val));
   }
 
   public static ExpressionContext forIdentifier(String identifier) {
-    return new ExpressionContext(Type.IDENTIFIER, identifier, null);
+    return new ExpressionContext(Type.IDENTIFIER, identifier, null, null);
   }
 
   public static ExpressionContext forFunction(FunctionContext function) {
-    return new ExpressionContext(Type.FUNCTION, null, function);
+    return new ExpressionContext(Type.FUNCTION, null, function, null);
   }
 
-  private ExpressionContext(Type type, String value, FunctionContext function) {
+  private ExpressionContext(Type type, String value, FunctionContext function, LiteralContext literal) {
     _type = type;
     _value = value;
     _function = function;
+    _literal = literal;
+  }
+
+  @Deprecated
+  public String getLiteralString() {
+    if (_literal == null || _literal.getValue() == null) {
+      return "";
```

Comment:
Should return `null` to keep the same behavior
```suggestion
      return null;
```

Reply:
Done.